### ISSUE TITLE
[Catch2] build failure

### ISSUE THREAD

**Issue by sibiarunachalam** [microsoft/vcpkg#24640]
**Host Environment**
- OS: Windows 10
- Compiler: VS 2019 cmake

**To Reproduce**
`./vcpkg install catch2`

**Failure logs**
```
- `The following packages will be built and installed:
    catch2[core]:x86-windows -> 2.13.8 -- C:\jenkins-root-dir\workspace\tar_Development_nectar-cpp_PR-71\cosmos\libs\vcpkg\buildtrees\versioning_\versions\catch2\b58473cdc953ae1d09f30f0ba1e641c6cc34719f
  * vcpkg-cmake[core]:x64-windows -> 2022-05-06 -- C:\jenkins-root-dir\workspace\tar_Development_nectar-cpp_PR-71\cosmos\libs\vcpkg\buildtrees\versioning_\versions\vcpkg-cmake\46a5686341cf9136053ff01b9f213dbd2dbea8a6
  * vcpkg-cmake-config[core]:x64-windows -> 2022-02-06 -- C:\jenkins-root-dir\workspace\tar_Development_nectar-cpp_PR-71\cosmos\libs\vcpkg\buildtrees\versioning_\versions\vcpkg-cmake-config\24dc7dfc704406e9f745f033643dc25f56e4ca18
Additional packages (*) will be modified to complete this operation.
A suitable version of 7zip was not found (required v21.7.0). Downloading portable 7zip v21.7.0...
Downloading 7zip...
  https://www.7-zip.org/a/7z2107-extra.7z -> C:\jenkins-root-dir\workspace\tar_Development_nectar-cpp_PR-71\cosmos\libs\vcpkg\downloads\7z2107-extra.7z
Extracting 7zip...
Restored 2 packages from C:\Users\Jenkins\AppData\Local\vcpkg\archives in 2.538 s. Use --debug to see more details.
Starting package 1/3: vcpkg-cmake-config:x64-windows
Installing package vcpkg-cmake-config[core]:x64-windows...
Elapsed time for package vcpkg-cmake-config:x64-windows: 155.7 ms
Starting package 2/3: vcpkg-cmake:x64-windows
Installing package vcpkg-cmake[core]:x64-windows...
Elapsed time for package vcpkg-cmake:x64-windows: 105.8 ms
Starting package 3/3: catch2:x86-windows
Building package catch2[core]:x86-windows...
-- Installing port from location: C:\jenkins-root-dir\workspace\tar_Development_nectar-cpp_PR-71\cosmos\libs\vcpkg\buildtrees\versioning_\versions\catch2\b58473cdc953ae1d09f30f0ba1e641c6cc34719f
-- Downloading https://github.com/catchorg/Catch2/archive/v2.13.8.tar.gz -> catchorg-Catch2-v2.13.8.tar.gz...
-- Extracting source C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/catchorg-Catch2-v2.13.8.tar.gz
-- Applying patch fix-install-path.patch
-- Using source at C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/src/v2.13.8-a12deb77cc.clean
-- Configuring x86-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe -v
    Working Directory: C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\jenkins-root-dir\workspace\tar_Development_nectar-cpp_PR-71\cosmos\libs\vcpkg\buildtrees\catch2\config-x86-windows-out.log

Call Stack (most recent call first):
  C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg_installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:331 (vcpkg_execute_required_process)
  buildtrees/versioning_/versions/catch2/b58473cdc953ae1d09f30f0ba1e641c6cc34719f/portfile.cmake:13 (vcpkg_cmake_configure)
  scripts/ports.cmake:146 (include)


error: building catch2:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+catch2
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[catch2]+Build+error
including:
  package: catch2[core]:x86-windows -> 2.13.8
    vcpkg-tool version: 2022-03-30-692785ac944e81417840c6de244fb3e18a4b35eb
    vcpkg-scripts version: b57434155 2022-05-10 (5 hours ago)
```
Additionally, attach any relevant sections from the log files above.
        1 dir(s) moved.
Applied user-wide integration for this vcpkg root.

All MSBuild C++ projects can now #include any installed libraries.
Linking will be handled automatically.
Installing new libraries will make them instantly available.

CMake projects should use: "-DCMAKE_TOOLCHAIN_FILE=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/scripts/buildsystems/vcpkg.cmake"`

-  **CMakeOutput.log**
```
`The system is: Windows - 10.0.19043 - AMD64
Compiling the CXX compiler identification source file "CMakeCXXCompilerId.cpp" succeeded.
Compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x86/cl.exe 
Build flags: ;/nologo;/DWIN32;/D_WINDOWS;/W3;/utf-8;/GR;/EHsc;/MP;
Id flags:  

The output was:
0
CMakeCXXCompilerId.cpp


Compilation of the CXX compiler identification source "CMakeCXXCompilerId.cpp" produced "CMakeCXXCompilerId.exe"

Compilation of the CXX compiler identification source "CMakeCXXCompilerId.cpp" produced "CMakeCXXCompilerId.obj"

The CXX compiler identification is MSVC, found in "C:/jenkins-dir/workspace/cpp-master/libs/vcpkg/buildtrees/catch2/x86-windows-dbg/CMakeFiles/3.22.2/CompilerIdCXX/CMakeCXXCompilerId.exe"

Detecting CXX compiler ABI info compiled with the following output:
Change Dir: C:/jenkins-dir/workspace/cpp-master/libs/vcpkg/buildtrees/catch2/x86-windows-dbg/CMakeFiles/CMakeTmp

Run Build Command(s):C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe cmTC_20b5c && [1/2] Building CXX object CMakeFiles\cmTC_20b5c.dir\CMakeCXXCompilerABI.cpp.obj

[2/2] Linking CXX executable cmTC_20b5c.exe
`
- **CMakeError.log**
 `Detecting CXX compiler ABI info failed to compile with the following output:
Change Dir: C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/x86-windows-dbg/CMakeFiles/CMakeTmp

Run Build Command(s):C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe cmTC_d1f48 && [1/2] Building CXX object CMakeFiles\cmTC_d1f48.dir\CMakeCXXCompilerABI.cpp.obj

[2/2] Linking CXX executable cmTC_d1f48.exe

FAILED: cmTC_d1f48.exe 

cmd.exe /C "cd . && C:\jenkins-root-dir\workspace\tar_Development_nectar-cpp_PR-71\cosmos\libs\vcpkg\downloads\tools\cmake-3.22.2-windows\cmake-3.22.2-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\cmTC_d1f48.dir --rc=rc --mt=CMAKE_MT-NOTFOUND --manifests  -- C:\PROGRA~2\MICROS~1.0\VC\bin\AMD64_~2\link.exe /nologo CMakeFiles\cmTC_d1f48.dir\CMakeCXXCompilerABI.cpp.obj  /out:cmTC_d1f48.exe /implib:cmTC_d1f48.lib /pdb:cmTC_d1f48.pdb /version:0.0 /machine:X86  /debug /INCREMENTAL /subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."

RC Pass 1: command "rc /fo CMakeFiles\cmTC_d1f48.dir/manifest.res CMakeFiles\cmTC_d1f48.dir/manifest.rc" failed (exit code 0) with the following output:
The system cannot find the file specified

ninja: build stopped: subcommand failed.





Determining if the CXX compiler works failed with the following output:
Change Dir: C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/x86-windows-dbg/CMakeFiles/CMakeTmp

Run Build Command(s):C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe cmTC_7ec9b && [1/2] Building CXX object CMakeFiles\cmTC_7ec9b.dir\testCXXCompiler.cxx.obj

[2/2] Linking CXX executable cmTC_7ec9b.exe

FAILED: cmTC_7ec9b.exe 

cmd.exe /C "cd . && C:\jenkins-root-dir\workspace\tar_Development_nectar-cpp_PR-71\cosmos\libs\vcpkg\downloads\tools\cmake-3.22.2-windows\cmake-3.22.2-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\cmTC_7ec9b.dir --rc=rc --mt=CMAKE_MT-NOTFOUND --manifests  -- C:\PROGRA~2\MICROS~1.0\VC\bin\AMD64_~2\link.exe /nologo CMakeFiles\cmTC_7ec9b.dir\testCXXCompiler.cxx.obj  /out:cmTC_7ec9b.exe /implib:cmTC_7ec9b.lib /pdb:cmTC_7ec9b.pdb /version:0.0 /machine:X86  /debug /INCREMENTAL /subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."

RC Pass 1: command "rc /fo CMakeFiles\cmTC_7ec9b.dir/manifest.res CMakeFiles\cmTC_7ec9b.dir/manifest.rc" failed (exit code 0) with the following output:
The system cannot find the file specified

ninja: build stopped: subcommand failed.
```

**Comment by Cheney-W** [comment]
Could you please provide us with `C:\jenkins-root-dir\workspace\tar_Development_nectar-cpp_PR-71\cosmos\libs\vcpkg\buildtrees\catch2\config-x86-windows-out.log`?

**Comment by sibiarunachalam** [comment]
Thanks for quick reply, attached the log.

**config-x86-windows-out.log**
`[1/2] cmd /c "cd .. && "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/src/570e89786c-dc421bea1a.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/packages/catch2_x86-windows" "-DBUILD_TESTING=OFF" "-DCATCH_BUILD_EXAMPLES=OFF" "-DCATCH_INSTALL_DOCS=OFF" "-DCATCH_BUILD_STATIC_LIBRARY=0" "-DCMAKE_MAKE_PROGRAM=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF""
FAILED: ../CMakeCache.txt 
cmd /c "cd .. && "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/src/570e89786c-dc421bea1a.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/packages/catch2_x86-windows" "-DBUILD_TESTING=OFF" "-DCATCH_BUILD_EXAMPLES=OFF" "-DCATCH_INSTALL_DOCS=OFF" "-DCATCH_BUILD_STATIC_LIBRARY=0" "-DCMAKE_MAKE_PROGRAM=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF""
-- The CXX compiler identification is MSVC 19.0.24245.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64_x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64_x86/cl.exe - broken
CMake Error at C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/share/cmake-3.22/Modules/CMakeTestCXXCompiler.cmake:62 (message):
  The C++ compiler

    "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64_x86/cl.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/x86-windows-rel/CMakeFiles/CMakeTmp
    
    Run Build Command(s):C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe cmTC_c13f8 && [1/2] Building CXX object CMakeFiles\cmTC_c13f8.dir\testCXXCompiler.cxx.obj
    [2/2] Linking CXX executable cmTC_c13f8.exe
    FAILED: cmTC_c13f8.exe 
    cmd.exe /C "cd . && C:\jenkins-root-dir\workspace\tar_Development_nectar-cpp_PR-71\cosmos\libs\vcpkg\downloads\tools\cmake-3.22.2-windows\cmake-3.22.2-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\cmTC_c13f8.dir --rc=rc --mt=CMAKE_MT-NOTFOUND --manifests  -- C:\PROGRA~2\MICROS~1.0\VC\bin\AMD64_~2\link.exe /nologo CMakeFiles\cmTC_c13f8.dir\testCXXCompiler.cxx.obj  /out:cmTC_c13f8.exe /implib:cmTC_c13f8.lib /pdb:cmTC_c13f8.pdb /version:0.0 /machine:X86  /debug /INCREMENTAL /subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
    RC Pass 1: command "rc /fo CMakeFiles\cmTC_c13f8.dir/manifest.res CMakeFiles\cmTC_c13f8.dir/manifest.rc" failed (exit code 0) with the following output:
    The system cannot find the file specified
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:19 (project)


-- Configuring incomplete, errors occurred!
See also "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/x86-windows-rel/CMakeFiles/CMakeOutput.log".
See also "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/x86-windows-rel/CMakeFiles/CMakeError.log".
[2/2] cmd /c "cd ../../x86-windows-dbg && "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/src/570e89786c-dc421bea1a.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/packages/catch2_x86-windows/debug" "-DBUILD_TESTING=OFF" "-DCATCH_BUILD_EXAMPLES=OFF" "-DCATCH_INSTALL_DOCS=OFF" "-DCATCH_BUILD_STATIC_LIBRARY=0" "-DCMAKE_MAKE_PROGRAM=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF""
FAILED: ../../x86-windows-dbg/CMakeCache.txt 
cmd /c "cd ../../x86-windows-dbg && "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/src/570e89786c-dc421bea1a.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/packages/catch2_x86-windows/debug" "-DBUILD_TESTING=OFF" "-DCATCH_BUILD_EXAMPLES=OFF" "-DCATCH_INSTALL_DOCS=OFF" "-DCATCH_BUILD_STATIC_LIBRARY=0" "-DCMAKE_MAKE_PROGRAM=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v140" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg_installed" "-DVCPKG_MANIFEST_INSTALL=OFF""
-- The CXX compiler identification is MSVC 19.0.24245.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - failed
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64_x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64_x86/cl.exe - broken
CMake Error at C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/share/cmake-3.22/Modules/CMakeTestCXXCompiler.cmake:62 (message):
  The C++ compiler

    "C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64_x86/cl.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/x86-windows-dbg/CMakeFiles/CMakeTmp
    
    Run Build Command(s):C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe cmTC_92b23 && [1/2] Building CXX object CMakeFiles\cmTC_92b23.dir\testCXXCompiler.cxx.obj
    [2/2] Linking CXX executable cmTC_92b23.exe
    FAILED: cmTC_92b23.exe 
    cmd.exe /C "cd . && C:\jenkins-root-dir\workspace\tar_Development_nectar-cpp_PR-71\cosmos\libs\vcpkg\downloads\tools\cmake-3.22.2-windows\cmake-3.22.2-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=CMakeFiles\cmTC_92b23.dir --rc=rc --mt=CMAKE_MT-NOTFOUND --manifests  -- C:\PROGRA~2\MICROS~1.0\VC\bin\AMD64_~2\link.exe /nologo CMakeFiles\cmTC_92b23.dir\testCXXCompiler.cxx.obj  /out:cmTC_92b23.exe /implib:cmTC_92b23.lib /pdb:cmTC_92b23.pdb /version:0.0 /machine:X86  /debug /INCREMENTAL /subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
    RC Pass 1: command "rc /fo CMakeFiles\cmTC_92b23.dir/manifest.res CMakeFiles\cmTC_92b23.dir/manifest.rc" failed (exit code 0) with the following output:
    The system cannot find the file specified
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:19 (project)


-- Configuring incomplete, errors occurred!
See also "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/x86-windows-dbg/CMakeFiles/CMakeOutput.log".
See also "C:/jenkins-root-dir/workspace/tar_Development_nectar-cpp_PR-71/cosmos/libs/vcpkg/buildtrees/catch2/x86-windows-dbg/CMakeFiles/CMakeError.log".
ninja: build stopped: subcommand failed.
`

**Comment by Cheney-W** [comment]
I can't reproduce this issue in my side with the latest vcpkg and VS2019.
I found your config-x86-windows-out.log have following setting:
`"-DVCPKG_PLATFORM_TOOLSET=v140"`

I'm not sure if you modified the ".\vcpkg\triplets\x86-windows.cmake" or you just installed the VS2015 in your computer, for fix this issue, I think you could clone a new vcpkg to install this port and install the VS2019 on your computer.

**Comment by sibiarunachalam** [comment]
Thank you for your valuable response.
We tried setting up new environment with VS2019 and build the project, it built successfully.

_Additional info:_
1. We had some stuff that related to VS2015 in the old build environment.
2. The old build environment doesn't work only after updating latest "built-baseline" from microsoft/vcpkg in vcpkg.json.

**Comment by Cheney-W** [comment]
Thanks for your reply! I'm glad to know your issue has been solved.
